### ISSUE TITLE
Question about prediction function

### ISSUE THREAD

**Issue by kyungeuuun** [PKUAI26/STGCN-IJCAI-18#6]
Hi,
I tried to apply your model to my data, however, 
model_test function is not working in full data size option, 
because there are too many routes in my data.
So I want to evaluate test performance in small batches.
On the other hand, according to my trial, 
full data performance and small batch performance were different at all.

Could you help me with this issue? Thanks :)

**Comment by VeritasYin** [comment]
Hi,
You can try to isolate the model_test function during the training progress. Also, the problem you described is also related to the batch_size and your GPU memory. The released code is tested on PeMS-L of 1026 nodes with the batch size as 50. It approximately consumes 8G memory for 12 steps prediction on NVIDIA GeForce GTX 1080Ti. 
Thank you

**Comment by kyungeuuun** [comment]
In the model inference phase, I tried two options.
(1) Small batch (i.e. batch_size <dataset.get_len ('test')), 2. Full dataset (i.e. batch_size = dataset.get_len('test')).
In Option 1, changing the batch_size changed the inference result for the test set, which was also different from the result in Option 2.
For example, for a total of 2000 test datasets, I changed the batch_size to {1000, 1005, 1010, ...} and all the test performance were different. Could you explain this phenomenon?

**Comment by kyungeuuun** [comment]
e.g. For the case of dataset.get_len('test') = 1185,
1. when, batch_size = 1175,
Time Step 3: MAPE 0.278; MAE  7.176; RMSE 11.877.
Time Step 6: MAPE 0.301; MAE  7.885; RMSE 13.301.
Time Step 9: MAPE 0.303; MAE  7.953; RMSE 13.315.
Time Step 12: MAPE 0.306; MAE  8.015; RMSE 13.336.
Model Test Time 7.047 s
Testing model finished!

2. when, batch_size = 1185,
Time Step 3: MAPE 0.117; MAE  2.947; RMSE 4.673.
Time Step 6: MAPE 0.135; MAE  3.347; RMSE 5.404.
Time Step 9: MAPE 0.147; MAE  3.596; RMSE 5.869.
Time Step 12: MAPE 0.156; MAE  3.789; RMSE 6.219.
Model Test Time 7.252 s
Testing model finished!

**Comment by kyungeuuun** [comment]
I tried with your PeMS-M dataset, however, I found the same result.
(full test data size = 1325)
1. when, batch_size = 1175: 
Time Step 3: MAPE 0.087; MAE  3.527; RMSE 6.470.
Time Step 6: MAPE 0.112; MAE  4.384; RMSE 7.711.
Time Step 9: MAPE 0.131; MAE  4.930; RMSE 8.586.
Time Step 12: MAPE 0.149; MAE  5.411; RMSE 9.363.
Model Test Time 3.235 s
Testing model finished!

2. when, batch_size = 1325:
Time Step 3: MAPE 0.053; MAE  2.271; RMSE 4.105.
Time Step 6: MAPE 0.078; MAE  3.117; RMSE 5.786.
Time Step 9: MAPE 0.099; MAE  3.749; RMSE 6.986.
Time Step 12: MAPE 0.119; MAE  4.301; RMSE 7.993.
Model Test Time 3.253 s
Testing model finished!
